test/core/QCheck2_tests.ml

```diff
@@ -727,6 +762,16 @@ module Stats = struct
   let char_dist =
     Test.make ~name:"char code dist" ~count:500_000 ~stats:[("char code", Char.code)] Gen.char (fun _ -> true)
 
+  let bytes_len_tests =
+    let len = ("len",Bytes.length) in
+    [
+      Test.make ~name:"bytes_size len dist"      ~count:5_000 ~stats:[len] Gen.(bytes_size (int_range 5 10)) (fun _ -> true);
+      Test.make ~name:"bytes len dist"           ~count:5_000 ~stats:[len] Gen.bytes                         (fun _ -> true);
+      Test.make ~name:"bytes_of len dist"        ~count:5_000 ~stats:[len] Gen.(bytes_of (return 'a'))       (fun _ -> true);
+      Test.make ~name:"bytes_printable len dist" ~count:5_000 ~stats:[len] Gen.bytes_printable               (fun _ -> true);
+      Test.make ~name:"small_bytes len dist"     ~count:5_000 ~stats:[len] Gen.(small_bytes ~gen:char)(*ugh*)(fun _ -> true);
```

Comment:
Is there any reason for the `(*ugh*)`

Reply:
Not really apart from copy-pasting the `string` tests.
As, ``small_string` line 782 has this `(* ugh *)` too, I think this is related to the presence of the optional `gen` argument.

Reply:
Not really apart from copy-pasting the `string` tests.
As, ``small_string` line 782 has this `(* ugh *)` too, I think this is related to the presence of the optional `gen` argument.

Reply:
I am to blame for the `(*ugh*)` - and if I remember correctly(tm) the reason was precisely being forced to provide an "optional `char`".

Reply:
I've removed them, and fixed what caused them.

Reply:
The problem is that for 1-argument combinators, the type-checker gets confused if the single argument is optional.
In effect, the "optional argument" is actually not optional - but required (for the code to type check).
This is a known issue with some of the existing combinators (see, e.g., #161).

Note: Since this is not a problem for 2-,3-,...-argument combinators such as `bytes_size` that also take a regular argument - I would favor to keep its signature aligned with `string_size` (still taking an optional parameter). Otherwise, it would be a breaking API change, AFAICS.

For `Gen.small_bytes` (and `Gen.small_string`) one option would be to
- keep the argument to both but make it labelled and hence mandatory (like in #161)
- add combinators
  - `Gen.bytes_small : bytes t`
  - `Gen.bytes_small_of : char t -> bytes t`
- (and similarly for `Gen.small_string`)

This has the advantage of
- not breaking backward compatibility
- aligning generators better, so that everything `string` or `bytes`-related start with the prefix `string...` or `bytes...` - which make them easier to search for using IDE-feedback...

We should agree on such a design first though. Thoughts @vch9 @c-cube ?

Reply:
I am happy to put back the `~gen`. Backward compatibility is a big argument.